### DISCUSSION TITLE
BITAND(), BITOR(), BITXOR() BIF's?

### DISCUSSION THREAD

**Discussion by rvjansen** [adesutherland/CREXX#129] [Ideas]
Rex 3.0 had BITAND(), BITOR(), BITXOR() bifs, I gather because the operators were not there yet. Also, I think these might be useful for cREXX level B as bifs. In addition, we might boast compatibility with even the oldest execs people can find. What do we think? If yes, I'll add them to [https://github.com/adesutherland/CREXX/wiki/Rexx-levelb-bif-planning] (url)

**Comment by adesutherland** [comment]
Not sure - happy to include them but we may need different versions of the library?

**Comment by Peter-Jacob** [comment]
BITAND/BITOR/BITXOR are string functions. I never in my life used them, but I wonder if the implementation isn't relatively easy, by using the new IAND/IOR/IXOR instructions. This example illustrates how it could be done (no length control, no padding yet): 
```
s1='abcdef'
s2='123456'
i1=0
i2=0
rs=0
result=''

do i=0 to 5
   assembler strchar i1,s1,i
   assembler strchar i2,s2,i
   assembler iand rs,i1,i2
   assembler itos rs
   assembler appendchar result,rs
end
say result
```

**Reply by rvjansen** [reply]
they have been removed from VM, maybe around the time the operators were enhanced. I think they have a right to exist as BIF, and if easy to implement, umso besser.
